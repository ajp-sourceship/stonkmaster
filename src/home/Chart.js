import React, {useState} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  View,
  Text,
  StatusBar,
  Dimensions,
} from 'react-native';

import {
  Header,
  LearnMoreLinks,
  Colors,
  DebugInstructions,
  ReloadInstructions,
} from 'react-native/Libraries/NewAppScreen';
import {AreaChart, Grid} from 'react-native-svg-charts';
import * as shape from 'd3-shape';

export const Chart = props => {
  return (
    <View
      style={{
        margin: 12,
        borderRadius: 6,
        padding: 0,
      }}>
      <AreaChart
        style={{height: 400}}
        data={props.data}
        contentInset={{top: 30, bottom: 30}}
        curve={shape.curveNatural}
        svg={{fill: 'rgba(134, 65, 244, 0.8)'}}>
        <Grid />
      </AreaChart>
    </View>
  );
};
