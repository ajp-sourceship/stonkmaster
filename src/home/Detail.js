/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React, {useState} from 'react';
import {
  View,
  Text
} from 'react-native';

import globals from '../globals/Globals';
import {iOSUIKit} from 'react-native-typography';
export const Detail = props => {
  return (
    <View
      style={{
        backgroundColor: globals.Colors.Black_Mid,
        margin: 12,
        borderRadius: 6,
        padding: 8,
      }}>
      <Text style={iOSUIKit.title3EmphasizedWhite}>{props.cryptoType}</Text>
      <Text style={{...iOSUIKit.title3EmphasizedWhite, margin:10}}>SIGNAL - {props.signal}</Text>

      {props.orders.map(order => {
          return <View style={{flexDirection:'row', justifyContent:'space-between'}}>
          <Text style={iOSUIKit.title3EmphasizedWhite}>{order.type }</Text>
          <Text style={iOSUIKit.title3EmphasizedWhite}>{order.date}</Text>
          <Text style={iOSUIKit.title3EmphasizedWhite}>{order.price}</Text>
          </View>
      })}
    </View>
  );
};
