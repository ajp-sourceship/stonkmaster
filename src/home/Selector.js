import React, {useState} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  View,
  Text,
  StatusBar,
  Dimensions,
  TouchableOpacity,
} from 'react-native';

import {
  Header,
  LearnMoreLinks,
  Colors,
  DebugInstructions,
  ReloadInstructions,
} from 'react-native/Libraries/NewAppScreen';
import globals from '../globals/Globals';
import {iOSUIKit} from 'react-native-typography';
import {AreaChart, Grid} from 'react-native-svg-charts';
import * as shape from 'd3-shape';
import {Chart} from './Chart';

export const Selector = props => {
  return (
    <View
      style={{
        justifyContent: 'space-between',
        flexDirection: 'row',
        marginVertical: 6,
      }}>
      <TouchableOpacity testID="btcButton" onPress={() => props.selectCrypto('BTC')}>
        <Text
          style={{
            color: 'white',
            borderColor: 'green',
            borderWidth: 2,
            borderRadius: 6,
            paddingVertical: 4,
            paddingHorizontal: 8,
          }}>
          BTC
        </Text>
      </TouchableOpacity>
      <TouchableOpacity testID="etgButton" onPress={() => props.selectCrypto('ETH')}>
        <Text
          style={{
            color: 'white',
            borderColor: 'green',
            borderWidth: 2,
            borderRadius: 6,
            paddingVertical: 4,
            paddingHorizontal: 8,
          }}>
          ETH
        </Text>
      </TouchableOpacity>
      <TouchableOpacity testID="dogButton" onPress={() => props.selectCrypto('DOGE')}>
        <Text
          style={{
            color: 'white',
            borderColor: 'green',
            borderWidth: 2,
            borderRadius: 6,
            paddingVertical: 4,
            paddingHorizontal: 8,
          }}>
          DOGE
        </Text>
      </TouchableOpacity>
    </View>
  );
};
