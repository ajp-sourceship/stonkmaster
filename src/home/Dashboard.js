/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React, {useState} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  View,
  Text,
  StatusBar,
  Dimensions,
} from 'react-native';

import {
  Header,
  LearnMoreLinks,
  Colors,
  DebugInstructions,
  ReloadInstructions,
} from 'react-native/Libraries/NewAppScreen';
import globals from '../globals/Globals';
import {iOSUIKit} from 'react-native-typography';
import {AreaChart, Grid} from 'react-native-svg-charts';
import * as shape from 'd3-shape';
import {Chart} from './Chart';
import {Selector} from './Selector';
import {Detail} from './Detail';

export const Dashboard = props => {
  const [cryptoType, setCryptoType] = useState('');
  const [dataType, setDataType] = useState([]);
  const [signal, setSignal] = useState([]);
  const [orders, setOrders] = useState([]);

  const btcdata = {
    signal: 'SELL',
    chartData: [50, 10, 40, 95, -4, -24, 85, 91, 35, 53, -53, 24, 50, -20, -80],
    orders: [
      {type: 'buy', price: 12.64, date: '06/12/21'},
      {type: 'buy', price: 24.58, date: '02/09/21'},
      {type: 'sell', price: 28.65, date: '06/26/21'},
    ],
  };
  const ethData = {
    signal: 'BUY',
    chartData: [-50, -10, -80, -65, -4, -8, 10, 40, 60, 80, 72, 76, 78, 83, 88],
    orders: [
      {type: 'buy', price: 12.64, date: '06/24/21'},
      {type: 'buy', price: 24.58, date: '03/09/21'},
      {type: 'sell', price: 28.65, date: '06/26/21'},
    ],
  };
  const dogeData = {
    signal: 'SELL',
    chartData: [25, 30, 35, -4, -4, -20, 30, -12, -2, 40, 80, 95, 78, 64, 88],
    orders: [
      {type: 'buy', price: 10.22, date: '06/02/21'},
      {type: 'buy', price: 14.22, date: '05/09/21'},
      {type: 'sell', price: 20.48, date: '06/26/21'},
    ],
  };
  const selectCrypto = selCrypto => {
    setCryptoType(selCrypto);
    switch (true) {
      case selCrypto === 'ETH':
        setDataType(ethData.chartData);
        setSignal(ethData.signal);
        setOrders(ethData.orders);
        break;
      case selCrypto === 'BTC':
        setDataType(btcdata.chartData);
        setSignal(btcdata.signal);
        setOrders(btcdata.orders);
        break;
      case selCrypto === 'DOGE':
        setDataType(dogeData.chartData);
        setSignal(dogeData.signal);
        setOrders(dogeData.orders);
        break;
    }
  };

  return (
    <View style={{flex: 1, backgroundColor: globals.Colors.Black}}>
      <View
        style={{
          backgroundColor: globals.Colors.Black_Mid,
          margin: 12,
          borderRadius: 6,
          padding: 8,
        }}>
        <Text testID='titleText' style={iOSUIKit.title3EmphasizedWhite}>
          Stonk Master - {cryptoType}
        </Text>
        <Selector selectCrypto={selected => selectCrypto(selected)} />
        {cryptoType === '' ? (
          <Text
            style={{...iOSUIKit.title3EmphasizedWhite, textAlign: 'center'}}>
            Press a Button to get started
          </Text>
        ) : (
          <Chart data={dataType} />
        )}
      </View>

      {cryptoType === '' ? null : (
        <Detail signal={signal} cryptoType={cryptoType} orders={orders} />
      )}
    </View>
  );
};
