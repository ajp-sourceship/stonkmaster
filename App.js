/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  View,
  Text,
  StatusBar,
} from 'react-native';

import {
  Header,
  LearnMoreLinks,
  Colors,
  DebugInstructions,
  ReloadInstructions,
} from 'react-native/Libraries/NewAppScreen';
import {Dashboard} from './src/home/Dashboard';
console.disableYellowBox = true;
const App  = () => {
  return (
    <View style={{flex:1}}>
     <Dashboard/>
    </View>
  );
};

export default App;
