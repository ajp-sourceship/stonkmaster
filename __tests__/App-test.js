
import 'react-native';
import React from 'react';
import App from '../App';
import Chart from '../src/home/Chart'
import {create, act} from 'react-test-renderer'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { Dashboard } from '../src/home/Dashboard';

test('renders correctly', () => {
  renderer.create(<App />);
});

const tree = create(<Dashboard/>)
test('chart renders correctly', () => {
  expect(tree).toMatchSnapshot()
});


test('setting crypto type', () =>
{
  const button = tree.root.findByProps({testID: 'btcButton'}).props;
  act(() => button.onPress())
  
  const text = tree.root.findByProps({testID: 'titleText'}).props;
  expect(text.children).toEqual(['Stonk Master - ',  'BTC']);
})