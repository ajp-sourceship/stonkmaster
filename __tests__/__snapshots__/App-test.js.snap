// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chart renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#121212",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#2c2c2c",
        "borderRadius": 6,
        "margin": 12,
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "600",
          "letterSpacing": 0.75,
          "lineHeight": 25,
        }
      }
      testID="titleText"
    >
      Stonk Master - 
      
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 6,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="btcButton"
      >
        <Text
          style={
            Object {
              "borderColor": "green",
              "borderRadius": 6,
              "borderWidth": 2,
              "color": "white",
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            }
          }
        >
          BTC
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="etgButton"
      >
        <Text
          style={
            Object {
              "borderColor": "green",
              "borderRadius": 6,
              "borderWidth": 2,
              "color": "white",
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            }
          }
        >
          ETH
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="dogButton"
      >
        <Text
          style={
            Object {
              "borderColor": "green",
              "borderRadius": 6,
              "borderWidth": 2,
              "color": "white",
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            }
          }
        >
          DOGE
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "600",
          "letterSpacing": 0.75,
          "lineHeight": 25,
          "textAlign": "center",
        }
      }
    >
      Press a Button to get started
    </Text>
  </View>
</View>
`;
